### PR TITLE
Add additional logging for axon exceptions

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: synapse/axon.py [hunk 1]
```diff
@@ -690,7 +690,7 @@ def _findAxonClones(self):
                 waiter.wait(60)
 
             except Exception as e:  # pragma: no cover
-                logger.exception('Unhandled exception during _findAxonClones')
+                logger.exception('Axon %s encountered a unhandled exception during _findAxonClones', self.iden)
```
Comment (vertexmc) [inline synapse/axon.py]: Can we also log the actual exception?

Comment (vEpiphyte) [reply synapse/axon.py]: this does log the exception stack trace

Comment (invisig0th) [reply synapse/axon.py]: It's a nit, but i'd be for messages like this being a bit "tighter" and less prose-ish. maybe:
```
axon: %s (_findAxonClones)
```
I think the "encountered an unhandled exception" stuff is redundant considering the traceback ;)  Come to think of it, even _findAxonClones might be redundant.  Basically, axon: %s by itself is probably all we need unless there's some more specific info about the exception context.

Comment (vEpiphyte) [reply synapse/axon.py]: updated to be less prosy and more infoy so there is more information available for investigating issues

**Pull request merged**